forward43/kickstarter_scraper.py

```diff
@@ -52,22 +56,18 @@ def fetch_data(url):
 
         for i in range(num_projects):
 
-            # TODO: This should be handled better
-            # TODO: Move this code
-            # TODO: It should be able to handle other scenarios
-            details = {
-                'id'          : str(data['projects'][i]['id']),
-                'rank'        : str(i + 1),
-                'name'        : data['projects'][i]['name'],
-                'creator'     : data['projects'][i]['creator']['name'],
-                'goal'        : str(data['projects'][i]['goal']),
-                'pledged'     : str(data['projects'][i]['pledged']),
-                'state'       : data['projects'][i]['state'],
-                'backers'     : str(data['projects'][i]['backers_count']),
-                'launch_date' : time.strftime('%c', time.localtime(data['projects'][i]['launched_at'])),
-                'deadline'    : time.strftime('%c', time.localtime(data['projects'][i]['deadline'])),
-                'url'         : data['projects'][i]['urls']['web']['project'],
-            }
+            details = { 'rank' : str(i + 1) }
+
+            for att in ATTRIBUTES:
+                details[att] = data['projects'][i].get(att, 'n.a.')
```

Comment:
This'll overwrite the rank attribute for all details as `'n.a.'` , I'd say either remove rank from the `ATTRIBUTES` , or if you want it in the `ATTRIBUTES`, add the rank after this for loop :)

Reply:
Good catch! I removed it from ATTRIBUTES.